src/main/java/com/transferwise/openbanking/client/api/common/BaseClient.java

```diff
@@ -0,0 +1,46 @@
+package com.transferwise.openbanking.client.api.common;
+
+import com.transferwise.openbanking.client.error.ApiCallException;
+import com.transferwise.openbanking.client.json.JsonConverter;
+import com.transferwise.openbanking.client.json.JsonReadException;
+import lombok.AccessLevel;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestClientResponseException;
+import org.springframework.web.client.RestOperations;
+
+/**
+ * Base class for all API clients, containing common functionality.
+ */
+@RequiredArgsConstructor(access = AccessLevel.PROTECTED)
+@Slf4j
+public class BaseClient {
```

Comment:
Not making this abstract as PMD throws a warning if an abstract class has no abstract methods in it.